compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECombineStridedOps.cpp

```diff
@@ -84,6 +77,28 @@ struct CombineStridedOps
                                         targetMemspaceInt.value());
       sourceMaxNbDims = dmaDimConfig.sourceMaxNbDims;
       targetMaxNbDims = dmaDimConfig.targetMaxNbDims;
+    } else if (auto npuCircularDmaOp =
+                   dyn_cast<AMDAIE::NpuCircularDmaCpyNdOp>(op.getOperation())) {
+      LLVM_DEBUG(llvm::dbgs()
+                 << "npuCircularDmaOp: " << npuCircularDmaOp << "\n");
+      AMDAIE::NpuCircularDmaCpyNdOp nextNpuCircDmaOp =
+          findNextDmaOpWithSameConnection(npuCircularDmaOp, block);
+      if (!nextNpuCircDmaOp) return failure();
+      nextStridedOp = cast<AMDAIE::DoublyStridedOpInterface>(
+          nextNpuCircDmaOp.getOperation());
+
+      std::optional<uint8_t> sourceMemspaceInt =
+          npuCircularDmaOp.getSourceMemorySpaceAsUInt();
+      std::optional<uint8_t> targetMemspaceInt =
+          npuCircularDmaOp.getTargetMemorySpaceAsUInt();
+      if (!sourceMemspaceInt || !targetMemspaceInt) {
+        return rewriter.notifyMatchFailure(
+            npuCircularDmaOp, "expected a source and target memory space");
+      }
```

Comment:
There's a scope of code reuse here since the preceding block does the same thing (besides gather ops to be deleted).

So, something like :-

```
if (auto npuDmaOp = dyn_cast<AMDAIE::NpuDmaCpyNdOp>(op.getOperation())) {
   // Step 1. Gather ops to be deleted.
   // Step 2. Get nextStridedOp/sourceMaxNbDims/targetMaxNbDims
} else if (auto npuCircularDmaOp = dyn_cast<AMDAIE::NpuCircularDmaCpyNdOp>(op.getOperation()) {
   // Step 1. Get nextStridedOp/sourceMaxNbDims/targetMaxNbDims
}
```

I'd also delete https://github.com/nod-ai/iree-amd-aie/blob/d9fc43bf656244cc2d1eb01a7a909e2504365411/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECombineStridedOps.cpp#L53C41-L53C52 while at it.

Reply:
Done, I move the duplicated logic out of the if/else.